homeassistant/components/speedtestdotnet/button.py

```diff
@@ -0,0 +1,52 @@
+"""Support for Speedtest.net internet speed testing button."""
+from __future__ import annotations
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.components.speedtestdotnet import SpeedTestDataCoordinator
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION, ENTITY_CATEGORY_CONFIG
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import slugify
+
+from .const import ATTRIBUTION, DEFAULT_NAME, DOMAIN, SPEED_TEST_SERVICE
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Speedtestdotnet buttons."""
```

Comment:
For consistency, I would change this to be `speedtest.net`. Also button should be singular since you can only have one of them.